galsim/config/input.py

```diff
@@ -294,6 +298,50 @@ def GetInputObj(input_type, config, base, param_name):
     return base['input_objs'][input_type][num]
 
 
+valid_input_types = {}
+
+def RegisterInputType(input_type, init_func, types, kwargs_func=None, has_nobj=False, 
+                      file_scope=False, setup_func=None):
+    """Register an input type for use by the config apparatus.
+
+    @param input_type       The name of the type in config['input']
+    @param init_func        A function or class name to use to build the input object.
+    @param types            A list of value or object types that use this input type.
+                            These items will have their "current" values invalidated when the
+                            input object changes.
+    @param kwargs_func      A function to get the initialization kwargs if the regular
+                            _req, _opt, etc. kind of initialization will not work. The call
+                            signature is:
+                                kwargs, safe = GetKwargs(config, base)
+                            [default: None, which means use the regular initialization]
```

Comment:
A pointer to documentation on the "regular" kind of initialization would be helpful here (maybe a signature for `init_func`?)